### PR TITLE
feat(new-functions): migrate lower() to new expression framework

### LINKED ISSUE #6774: `lower`

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

migrate `lower()` to new expression framework

**example**

```
ast            : lower('DOBRÝ DEN')
raw expr       : lower(DOBRÝ DEN)
checked expr   : lower<String>(DOBRÝ DEN)
output type    : String
output domain  : {""..}
output         : dobrý den
```

Fixes #6774

### CODE REVIEW

**Comment by PsiACE** [comment]
`Dev Linux / build_aarch64_musl (pull_request) ` is OOM?

File: common/functions-v2/tests/it/scalars/string.rs [hunk 1]
```diff
@@ -42,6 +43,18 @@ fn test_upper(file: &mut impl Write) {
     )]);
 }
 
+fn test_lower(file: &mut impl Write) {
+    run_ast(file, "lower('Abc')", &[]);
+    run_ast(file, "lower('DOBRÝ DEN')", &[]);
+    run_ast(file, "lower('ß😀山')", &[]);
```
Comment (andylokandy) [inline common/functions-v2/tests/it/scalars/string.rs:49]: Let's try something fancy:

```suggestion
    run_ast(file, "lower('İ😀山')", &[]);
```

File: common/functions-v2/src/scalars/string.rs [hunk 2]
```diff
@@ -39,5 +39,27 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                        writer.put_char(x);
+                    }
+                }
+            }
+            writer.commit_row();
+            Ok(())
+        },
+    );
     registry.register_aliases("upper", &["ucase"]);
+    registry.register_aliases("lower", &["lcase"]);
```
Comment (andylokandy) [inline common/functions-v2/src/scalars/string.rs:64]: Same as https://github.com/datafuselabs/databend/pull/6780#discussion_r928151581

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
@mergify update

**Comment by andylokandy** [comment]
@PsiACE please fix conflicts

**Comment by andylokandy** [comment]
@mergify update

File: common/functions-v2/tests/it/scalars/string.rs [hunk 3]
```diff
@@ -44,6 +45,17 @@ fn test_upper(file: &mut impl Write) {
         build_string_column(&["Abc", "Dobrý den", "ß😀山"]),
     )]);
 }
+fn test_lower(file: &mut impl Write) {
```
Comment (andylokandy) [inline common/functions-v2/tests/it/scalars/string.rs:48]:
```suggestion

fn test_lower(file: &mut impl Write) {
```

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**